core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
Hm, yeah that anonymous user stuff does seem awkward.

It seems there's no way to really enforce FirstOrder since it's usually easy to create new accounts or use new email addresses.  FirstOrder just makes a best effort to remind users what the intention of the promotion is.  And thus it incorporates searching by email address. But NthOrder (for N > 1) does have some enforcement since users have to have placed some other orders before they make it to order N.  Users might be able to hack NthOrder by making+canceling orders but that's a lot harder than switching your email address.

So I guess it does seem like N=1 and N>1 have some fundamental differences and I'd vote to go ahead and keep them separate.